File: memories/2023/exercise/scala/day07/src/test/scala/CapturingLogger.scala
import ci.dependencies.Logger

import scala.collection.mutable.ListBuffer


class CapturingLogger extends Logger {
  private var lines: ListBuffer[String] = ListBuffer()

  def getLoggedLines: Seq[String] = lines.toList

  override def info(message: String): Unit = lines += "INFO: " + message

  override def error(message: String): Unit = lines += "ERROR: " + message

  def startCapture(): Unit = lines = ListBuffer()
}

File: memories/2023/exercise/php/day07/tests/CapturingLogger.php
<?php

declare(strict_types = 1);

namespace Tests;

use App\Dependancies\Logger;

class CapturingLogger implements Logger
{
	private array $lines = [];

	public function info(string $message): void
	{
		$this->lines[] = 'INFO: ' . $message;
	}

	public function error(string $message): void
	{
		$this->lines[] = 'ERROR: ' . $message;
	}

	public function getLoggedLines(): array
	{
		return $this->lines;
	}
}
